packages/nightly/src/lib/nightly.ts

```diff
@@ -0,0 +1,194 @@
+import type {
+  InjectedWallet,
+  WalletBehaviourFactory,
+  WalletModuleFactory,
+  WalletSelectorStore,
+} from "@near-wallet-selector/core";
+import { Account, waitFor } from "@near-wallet-selector/core";
+import { createAction } from "@near-wallet-selector/wallet-utils";
+import { isMobile } from "is-mobile";
+import { utils } from "near-api-js";
+import { AccessKeyView } from "near-api-js/lib/providers/provider";
+import { createTransaction } from "near-api-js/lib/transaction";
+import { PublicKey } from "near-api-js/lib/utils";
+import type { NearNightly, InjectedNightly } from "./injected-nightly";
+
+declare global {
+  interface Window {
+    nightly: InjectedNightly | undefined;
+  }
+}
+
+interface NightlyState {
+  wallet: NearNightly;
+}
+
+const setupNightlyState = async (
+  store: WalletSelectorStore
+): Promise<NightlyState> => {
+  const { selectedWalletId } = store.getState();
+  const wallet = window.nightly!.near!;
+  // Attempt to reconnect wallet if previously selected.
+  if (selectedWalletId === "nightly") {
+    await wallet.connect(undefined, true).catch(() => null);
+  }
+  return {
+    wallet,
+  };
+};
+const isInstalled = () => {
+  return waitFor(() => !!window.nightly!.near!).catch(() => false);
+};
+const Nightly: WalletBehaviourFactory<InjectedWallet> = async ({
+  metadata,
+  store,
+  logger,
+  provider,
+}) => {
+  const _state = await setupNightlyState(store);
+
+  const currentState = store.getState();
+  if (currentState.selectedWalletId === "nightly") {
+    try {
+      // eager connect to the wallet
+      await _state.wallet.connect(undefined, true);
+    } catch {
+      // ignore
+    }
+  }
+  const getAccounts = async () => {
+    if (!_state || _state.wallet.account.accountId === "") {
+      return [];
+    }
+    const nearAccount: Account = {
+      accountId: _state.wallet.account.accountId,
+    };
+    return [nearAccount];
+  };
+  return {
+    // nightly does not support delegating signing right now
+    async signIn() {
+      // If wallet does not exist user will be redirected to download page
+      if (!_state) {
+        window.location.href = metadata.downloadUrl;
+        throw new Error("Redirecting to download");
+      }
+      const existingAccount = _state.wallet.account.accountId;
+
+      if (existingAccount) {
+        const nearAccount: Account = {
+          accountId: _state.wallet.account.accountId,
+        };
+        return [nearAccount];
+      }
```

Comment:
We should reuse `getAccounts` here

Reply:
fix https://github.com/near/wallet-selector/pull/314/commits/45125a100f0dc1944e462a92180be28072cea983